added nod mission 2b

42foobar42 opened the pull request:
So this mission is very similar to 2a.

alexander-boll committed:
added nod mission 2b

obrakmann commented:
Looks good to me, :+1:

abcdefg30 commented:
We need something (just a hunt activity?) for the case that the base is at a different location:
![nod02b](https://cloud.githubusercontent.com/assets/7704140/6093336/bf002e82-aeff-11e4-8c79-cf315c805b5f.png)
(Image: Idling gdi units)

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+	local Harv = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+		return actor.Owner == GDI and actor.Type == 'harv' and not actor.IsDead
+	end)	
+	OnAnyDamaged({Harv[1]}, Atk5TriggerFunction)
```
Why that complex? ```Trigger.OnDamaged(Harv[1], Atk5TriggerFunction)``` should be enough.

pchote commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+end
+
+setTriggerAtk5 = function()
+	local Harv = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
```
This is pretty gross.  It would be better to suppress the harvester spawning (add `FreeActor: false` to the `Refinery: proc`) and then place/spawn the harvester manually.

Mailaender commented:
Yep, definitely needs an idle hunt trigger somewhere:

![image](https://cloud.githubusercontent.com/assets/756669/6095799/3bf92352-af6f-11e4-8f37-eceb779a1beb.png)

These guys are camping in front of my base, but not attacking.

abcdefg30 commented on mods/cnc/maps/nod02b/map.yaml:
```diff
@@ -0,0 +1,396 @@
+
+Title: Invasion of Egypt (b)
+
+Description: GDI has kept a stranglehold on Egypt for many years.  Set up a forward attack base in your area.  To do this you must select your Mobile Construction Vehicle (MCV) and right click on it. From here you can begin to build a base. This area contains plenty of Tiberium, so establishing the base should be easy.
```
You're missing a space (there should be two) between ```on it.``` and ```From``` in comparison to nod02a.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+Grd2TriggerFunction = function()
+	if not Grd2TriggerSwitch then
+		Grd2TriggerSwitch = true
+		MyActors = getActors(GDI, { ['e1'] = 5 })		
```
Two unintentional tabs at the end.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+		MyActors = getActors(GDI, { ['e1'] = 5 })		
+		Utils.Do(MyActors, function(actor)
+			Gdi5Movement(actor)
+		end) 
```
Space at the end.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+	MyActors = getActors(GDI, { ['e1'] = 2 })
+	Utils.Do(MyActors, function(actor)
+		Gdi1Movement(actor)
+	end) 
```
Same here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+	MyActors = getActors(GDI, { ['e1'] = 3 })
+	Utils.Do(MyActors, function(actor)
+		Gdi3Movement(actor)
+	end) 
```
And here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+setTriggerAtk5 = function()
+	local Harv = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+		return actor.Owner == GDI and actor.Type == 'harv' and not actor.IsDead
+	end)	
```
And here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+	end) 
+end
+
+Atk4TriggerFunction = function()	  
```
Undesired spaces at the end again.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+	end)
+end
+
+Atk5TriggerFunction = function()		
```
And here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+end
+
+Atk5TriggerFunction = function()		
+	if not Atk5TriggerSwitch then		
```
And here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+end
+
+Gdi5Movement = function(unit)
+	unit.Hunt()	
```
And here.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,230 @@
+			for i = 1, unitsToProduce, 1 do
+				UnitsType[i] = UnitToRebuild
+			end
+			Barracks.Build(UnitsType)			
```
And here.

42foobar42 commented:
I removed all tabs and spaces.

I added the harvester like it was in the original mission and deleted the obsolete lua code.

Also I added a trigger that all units go hunting when all buildings of the base are destroyed. I also added hunt to the end of every movement of the gdi units.

abcdefg30 commented on mods/cnc/maps/nod02b/nod02b.lua:
```diff
@@ -0,0 +1,236 @@
+
+	Trigger.AfterDelay(0, getStartUnits)
+	Harvester.FindResources()
+	--Trigger.AfterDelay(0, setTriggerAtk5)
```
Please remove this, if you won't use it.

abcdefg30 commented:
Otherwise :+1:.

42foobar42 commented:
Upps, forgot to delete...

Do you mean adding hunt to the end of every movement? Can I do this also with this PR?

abcdefg30 commented:
You can do it here, but if you wish you can submit another PR instead.

alexander-boll committed:
added hunt to every movement

42foobar42 commented:
added the hunting to map 2a

abcdefg30 commented:
Ah wait, sorry, my mistake.
It would be better to have an "idle-trigger". At the moment units will stop after the killed one enemy.

I solved this in e.g. allies03a this way:
```IdleHunt = function(unit) if not unit.IsDead then Trigger.OnIdle(unit, unit.Hunt) end end```
and then you'd need just ```IdleHunt(unit)``` instead of ```unit.Hunt()```.

42foobar42 commented:
No, problem.

Changed hunting as you recommended it.

obrakmann commented:
Looks fine to me, :+1: 

I'm going to merge this despite the Appveyor build failure, [which seems to be a problem on their end](https://ci.appveyor.com/project/OpenRA/openra/build/1.0.1001#L21) ("Unable to connect to the remote server").

obrakmann commented:
Thanks!

[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/fb578a9f4ccd231080c02156c8a08b2f36a4c80f%5E...fb578a9f4ccd231080c02156c8a08b2f36a4c80f)

42foobar42 commented:
Your welcome! Next mission will take a few days. 

So, see you soon ;)
